### PR TITLE
feat: add `chainhook docs api generate` command

### LINKED ISSUE #251: OpenAPI documentation
Investigate possible regression in https://raw.githubusercontent.com/hirosystems/chainhook/develop/docs/chainhook-openapi.json

### PR DESCRIPTION
**Pull request by MicaiahReid**
Fixes #251.

I've update `chainhook-openapi.json`, and I've added the `chainhook docs api generate` command to regenerate the spec as the api changes.

We should update the `chainhook-openapi.json` as part of our deployment process eventually.

Also, this introduces us using a fork of `schemars` and `okapi`. I've created PRs in this repo so we can hopefully use the stable version again before too long.

### CODE REVIEW

**Comment by lgalabru** [comment]
Thanks @MicaiahReid! did you fix the spec manually? Asking because I'd like to augment one of the core schema in another PR.

**Comment by MicaiahReid** [comment]
@lgalabru Yes, sadly I did do this manually. 

I've started looking into opening a PR to fix these two:
 - https://github.com/GREsau/schemars/issues/230
 - https://github.com/GREsau/okapi/issues/128

I think that would make the autogenerated schema correct

File: components/chainhook-cli/src/service/mod.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-mod http_api;
+pub(crate) mod http_api;
```
Comment (MicaiahReid) [inline components/chainhook-cli/src/service/mod.rs:1]: @lgalabru, I'm not sure if this is the right way to do this, but it works ¯\_(ツ)_/¯

File: components/chainhook-cli/Cargo.toml [hunk 2]
```diff
@@ -38,9 +38,13 @@ atty = "0.2.14"
 crossbeam-channel = "0.5.8"
 uuid = { version = "1.3.0", features = ["v4", "fast-rng"] }
 threadpool = "1.8.1"
-rocket_okapi = "0.8.0-rc.3"
+rocket_okapi = { version = "0.8.0-rc.3", git = "https://github.com/MicaiahReid/okapi.git", branch = "feat-chainhook-fixes"}
```
Comment (lgalabru) [inline components/chainhook-cli/Cargo.toml:41]: Could we host the fork under the `hirosystems` org?

File: components/chainhook-cli/src/service/mod.rs (hunk 1 above)
Comment (lgalabru) [reply components/chainhook-cli/src/service/mod.rs:1]: Yep that works!

File: components/chainhook-cli/Cargo.toml (hunk 2 above)
Comment (MicaiahReid) [reply components/chainhook-cli/Cargo.toml]: Why must you thwart my plans for github fame?

Comment (MicaiahReid) [reply components/chainhook-cli/Cargo.toml]: Fixed!

File: components/chainhook-sdk/Cargo.toml [hunk 3]
```diff
@@ -32,7 +32,11 @@ reqwest = { version = "0.11", default-features = false, features = [
 ] }
 tokio = { version = "1.24", features = ["full"] }
 base58 = "0.2.0"
-schemars = { version = "0.8.10" }
+clap = { version = "3.1.6", features = ["derive"], optional = true }
+clap_generate = { version = "3.0.3", optional = true }
+toml = { version = "0.5.6", features = ["preserve_order"], optional = true }
+ctrlc = { version = "3.2.2", optional = true }
+schemars = { version = "0.8.10", git = "https://github.com/hirosystems/schemars.git", branch = "feat-chainhook-fixes" }
```
Comment (lgalabru) [inline components/chainhook-sdk/Cargo.toml:39]:
```suggestion
schemars = { version = "0.8.10", git = "https://github.com/hirosystems/schemars.git", branch = "feat-chainhook-fixes" }
```

**Review by lgalabru — COMMENTED** [review]
Looking great! just a few adjustment on dependencies.

**Pull request merged**